test/performance/operf.go

```diff
@@ -0,0 +1,634 @@
+	filesystemTemplate = `sum(pod:container_fs_usage_bytes:sum * on(pod) group_left(workload, workload_type) namespace_workload_pod:kube_pod_owner:relabel{workload="{{.Name}}", workload_type="{{.WorkloadType}}"}) by (pod)`
+	Mebibyte           = 1048576 // bytes
+)
+
+const (
+	usageHelp = `Usage of operf:
+This script provide LVM operator metrics retrieval (for all the units in the operator) doing two different kind of tests:
+
+- LVMcluster creation:
+	This implies deployment of daemonsets in the nodes selected and the LVM Volume Groups creation.
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -lvmcrd config/samples/lvm_v1alpha1_lvmcluster.yaml
+
+- PVCs and PVs creation and usage:
+	It is created <number> of "busy" pods using PVCs bounded to PVs which use the Storage class provided
+	Example:
+	# go run test/performance/operf.go -token sha256~cj81ClyUYu7g05y8K-uLWm2AbrKTbNEQ96hEJcWStQo -number 16
+
+Parameters:`
+)
+
+var (
+	// flags
+	testNamespace        *string
+	testStorageClassName *string
+	namePattern          *string
+	numberInstances      *int
+	lvmClusterFile       *string
+	token                *string
+
+	// variables
+	PVCLabels     = map[string]string{"app": "testPerformance"}
+	prometheusURL string
+)
+
+func main() {
+	testNamespace = flag.String("namespace", "lvm-operator-system", "Namespace where the PVCs and test pods will be deployed/undeployed")
+	testStorageClassName = flag.String("sc", "odf-lvm-vg1", "Name of the topolvm storage class that will be used in the PVCs")
+	namePattern = flag.String("pattern", "perfotest", "Pattern used to build the PVCs/Pods names")
+	numberInstances = flag.Int("number", 4, "Number of Pods/Pvcs (each Pod uses a different PVC) to create in the test")
+	lvmClusterFile = flag.String("lvmcrd", "", "CRD file for the lvm cluster. Using this parameter will only execute the <LVMcluster creation> (VGs and SC creation)")
+	token = flag.String("token", "", "Mandatory authentication token needed to connect with the Openshift cluster")
+
+	flag.Usage = func() {
+		flagSet := flag.CommandLine
+		fmt.Println(usageHelp)
+		order := []string{"namespace", "pattern", "number", "sc", "lvmcrd", "token"}
+		for _, name := range order {
+			flag := flagSet.Lookup(name)
+			fmt.Printf("\t-%s (default: %s)\t%s\n", flag.Name, flag.DefValue, flag.Usage)
+		}
+	}
+
+	flag.Parse()
+
+	if *token == "" {
+		fmt.Println("Please provide the token parameter to connect with the Openshift cluster. Aborting test.")
+		return
+	}
+
+	var kubeconfig *string
+	val, ok := os.LookupEnv("KUBECONFIG")
+	if !ok {
+		panic("KUBECONFIG env var not defined")
+	}
+	kubeconfig = &val
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
```

Comment:
There is [deploymanager](https://github.com/red-hat-storage/lvm-operator/blob/e770bcb4b3793edc353a1d4415741b027a1561ce/pkg/deploymanager/config.go#L55) created for e2e tests. 
Can that be reused here?

Reply:
We can do that, But we need to make sure that we do not change anything in the future which will break this script. Moreover, it is 15 lines of code while replacing it we are hardly able to save a few lines of code.

Reply:
> We can do that, But we need to make sure that we do not change anything in the future which will break this script. 
Agreed. But this mostly looks like static code to get the client which I don't expect to change.

> Moreover, it is 15 lines of code while replacing it we are hardly able to save a few lines of code.
The idea is the have a common a utility that will provide the client to connect with k8s via kubeconfig.  In future, we need to create the client again, then this utility can be reused, rather than writing same logic again to generate the client.

Reply:
I just tried this, The deploy manager uses the client from `"sigs.k8s.io/controller-runtime/pkg/client"` and this script uses the client from `"k8s.io/client-go/kubernetes"`. We can not directly use the deploy manager client here directly.